core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentStateController.java

```diff
@@ -132,12 +132,16 @@ private void started() {
         initialLoadingInfo.setOperationStatus(WS_AGENT_BOOTING.getValue(), SUCCESS);
         loader.hide();
 
-        if (messageBusCallback != null) {
-            messageBusCallback.onSuccess(messageBus);
-        }
-        if (devMachineCallback != null) {
-            devMachineCallback.onSuccess(devMachine);
+        for (AsyncCallback<MessageBus> callback : messageBusCallbacks) {
+        	callback.onSuccess(messageBus);
+		}
+        messageBusCallbacks = null;
```

Comment:
It should. 'null' is the protocol for "no more callbacks". In case there is a bug elsewhere we will have an NPW here. Much better than silently ignoring callbacks which will never be called.